### PR TITLE
fix: add mfa reference docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Add MFA methods to reference docs

### CODE REVIEW

File: spec/common-client-libs-sections.json
Comment (dannykng) [inline spec/common-client-libs-sections.json:607]: Should this be:
```suggestion
        "title": "Create and verify a challenge",
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:461]: Incomplete sentence?

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:469]:
```suggestion
        name: Enroll a time-based, one-time password (TOTP) factor
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:482]:
```suggestion
        name: Create a challenge for a factor
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:495]:
```suggestion
        name: Verify a challenge for a factor
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:512]:
```suggestion
        name: Create and verify a challenge for a factor
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:527]:
```suggestion
      - In Supabase, having an AAL of `aal1` refers to having the 1st factor of authentication such as an email and password or OAuth sign-in while `aal2` refers to the 2nd factor of authentication such as a time-based, one-time-password (TOTP).
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:821]: For consistency:
```suggestion
        name: List all factors for a user
```

File: spec/supabase_js_v2_temp_new_shape.yml
Comment (dannykng) [inline spec/supabase_js_v2_temp_new_shape.yml:832]:
```suggestion
        name: Delete a factor for a user
```

**Review by dannykng — APPROVED** [review]

**Comment by MildTomato** [comment]
Screenshot 2022-12-13 at 12 31 50 PM

make sure Docs has built preview before merging

**Comment by MildTomato** [comment]
Screenshot 2022-12-13 at 12 46 56 PM

we also need to be vigilent about these enrichment changes.
make sure they are actually new updated ones when they change

**Review by MildTomato — APPROVED** [review]

**Comment by MildTomato** [comment]
Whats going on with these enrichments? 
@kangmingtay are you running `make` in the `/spec` directory ? or copying things over manually ?

**Review by MildTomato — CHANGES_REQUESTED** [review]
blocked merging while clarifying enrichment issue

**Review by MildTomato — APPROVED** [review]

**Comment by MildTomato** [comment]
snuck missing env vars for ticket site in

**Pull request merged**
